From 76950d08a3da13dd9e51b2fbef3532c3de79265b Mon Sep 17 00:00:00 2001
From: Dominique Martinet <asmadeus@codewreck.org>
Date: Fri, 14 Jun 2024 23:19:05 +0900
Subject: [PATCH] utils: fix kernel headers not found warning (#3242)

The code would make ksrc and kobj empty then try to print them,
this would always print empty strings.
Store is_dir() checks as bool instead and use these, the strings
cannot be empty if the check passed.

(cherry picked from commit f937803c2ad156ab90b9194965dbfb62bef1ff80)
---
(mostly to avoid conflicts with the next patch)
 src/utils.cpp | 21 +++++++--------------
 1 file changed, 7 insertions(+), 14 deletions(-)

diff --git a/src/utils.cpp b/src/utils.cpp
index bda722136588..c358a401fc83 100644
--- a/src/utils.cpp
+++ b/src/utils.cpp
@@ -738,15 +738,11 @@ std::tuple<std::string, std::string> get_kernel_dirs(
   auto ksrc = kdir + "/source";
   auto kobj = kdir + "/build";
 
-  // if one of source/ or build/ is not present - try to use the other one for both.
-  if (!is_dir(ksrc)) {
-    ksrc = "";
-  }
-  if (!is_dir(kobj)) {
-    kobj = "";
-  }
-  if (ksrc.empty() && kobj.empty())
-  {
+  // if one of source/ or build/ is not present - try to use the other one for
+  // both.
+  auto has_ksrc = is_dir(ksrc);
+  auto has_kobj = is_dir(kobj);
+  if (!has_ksrc && !has_kobj) {
     LOG(WARNING) << "Could not find kernel headers in " << ksrc << " or "
                  << kobj
                  << ". To specify a particular path to kernel headers, set the "
@@ -757,12 +753,9 @@ std::tuple<std::string, std::string> get_kernel_dirs(
                     "file at /sys/kernel/kheaders.tar.xz";
     return std::make_tuple("", "");
   }
-  if (ksrc.empty())
-  {
+  if (!has_ksrc) {
     ksrc = kobj;
-  }
-  else if (kobj.empty())
-  {
+  } else if (!has_kobj) {
     kobj = ksrc;
   }
 
-- 
2.45.2

